util/tune/current_settings.rb

```diff
@@ -0,0 +1,366 @@
+#
+# @author Bill Claytor
+#
+# @param [String] file The file to search for java args
+#
+# @return [string] The java args
+#
+# @example
+#   args = get_java_args(file)
+#
+# TODO: grep for java args
+def get_java_args(file)
+  path = if File.exist? "/etc/debian_version"
+           "/etc/defaults/#{file}"
+         else
+           "/etc/sysconfig/#{file}"
+         end
+
+  value = if File.exist? path
+            File.open(path).grep(/Xmx/)[0].split('"')[1]
+          else
+            NA
+          end
+
+  value
+end
+
+# Returns the value for the following setting:
+# "puppet_enterprise::master::puppetserver::reserved_code_cache"
+#
+# Uses the 'ReservedCodeCacheSize' parameter in the 'pe-puppetserver' file
+#
+# @author Bill Claytor
+#
+# @return [string] The reserved_code_cache value
+#
+# @example
+#   value = puppetserver_reserved_code_cache
+#
+#
+def puppetserver_reserved_code_cache
+  get_java_args("pe-puppetserver").match(/XX:ReservedCodeCacheSize=\K[^\s]+/) || NA
+end
+
+# Returns the value for the following setting:
+# 'puppet_enterprise::profile::console::java_args'
+#
+# Uses the 'pe-console-services' file
+#
+# @author Bill Claytor
+#
+# @return [string] The java args
+#
+# @example
+#   value = console_java_args
+#
+def console_java_args
+  get_java_args("pe-console-services")
+end
+
+# Returns the value for the following setting:
+# 'puppet_enterprise::profile::master::java_args'
+#
+# Uses the 'pe-puppetserver' file
+#
+# @author Bill Claytor
+#
+# @return [string] The java args
+#
+# @example
+#   value = master_java_args
+#
+def master_java_args
+  get_java_args("pe-puppetserver")
+end
+
+# Returns the value for the following setting:
+# 'puppet_enterprise::profile::orchestrator::java_args'
+#
+# Uses the 'pe-orchestration-services' file
+#
+# @author Bill Claytor
+#
+# @return [string] The java args
+#
+# @example
+#   value = zzz
+#
+def orchestrator_java_args
+  get_java_args("pe-orchestration-services")
+end
+
+# Returns the value for the following setting:
+# 'puppet_enterprise::profile::puppetdb::java_args'
+#
+# Uses the 'pe-puppetdb' file
+#
+# @author Bill Claytor
+#
+# @return [string] The java args
+#
+# @example
+#   value = puppetdb_java_args
+#
+def puppetdb_java_args
+  get_java_args("pe-puppetdb")
+end
+
+# Returns the value for the specified parameter in the specified config file
+# Optionally accepts a list of exclusions to avoid collisions (i.e. 'work_mem' matching 'autovacuum_work_mem', etc)
+#
+# @author Bill Claytor
+#
+# @param [String] file The config file to search
+# @param [String] parameter The parameter to search for
+# @param [RegEx]  exclusions The regular expression used to exclude words from the search
+#
+# @return [string] The parameter value if found, otherwise "N/A"
+#
+# @example
+#   value = get_conf_parameter(file, parameter)
+#   value = get_conf_parameter(file, parameter, /(example_a|example_b)/)
+#
+def get_conf_parameter(file, parameter, exclusions = NA)
```

Comment:
The use of NA here is weird.  Cause it just passed the value of NA as an exclusion... what I would kinda expect is that it wouldn't do a grep_v on exclusions if there aren't any.

Reply:
I used this to show that it was not applicable (and should not cause an undesired exclusion). It works with `exclusions = ""` if that is preferable, or I can do an if / else which is actually what this replaced :)